From c8944a2bf81019ed2026d1e0ca741424f81c8be2 Mon Sep 17 00:00:00 2001
From: Lasse Laukkanen <ext-lasse.2.laukkanen@nokia.com>
Date: Wed, 11 Mar 2009 18:31:17 +0200
Subject: [PATCH 02/23] camerabin: change default yuv format from I420 to UYVY

---
 gst/camerabin/gstcamerabin.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst/camerabin/gstcamerabin.c b/gst/camerabin/gstcamerabin.c
index a17e02c..7405dc3 100644
--- a/gst/camerabin/gstcamerabin.c
+++ b/gst/camerabin/gstcamerabin.c
@@ -204,7 +204,7 @@ static guint camerabin_signals[LAST_SIGNAL];
 #define DEFAULT_FPS_N 0         /* makes it use the default */
 #define DEFAULT_FPS_D 1
 
-#define CAMERABIN_DEFAULT_VF_CAPS "video/x-raw-yuv,format=(fourcc)I420"
+#define CAMERABIN_DEFAULT_VF_CAPS "video/x-raw-yuv,format=(fourcc)UYVY"
 #define CAMERABIN_MAX_VF_WIDTH 848
 #define CAMERABIN_MAX_VF_HEIGHT 848
 
-- 
1.7.10.4

