From 20d9b002af403d85afc9e9a9eac87dcbe799b780 Mon Sep 17 00:00:00 2001
From: Lasse Laukkanen <ext-lasse.2.laukkanen@nokia.com>
Date: Fri, 24 Sep 2010 18:01:25 +0300
Subject: [PATCH] camerabin: make sure that preview caps is properly set to video source

---
 gst/camerabin/gstcamerabin.c |   36 ++++++++++++++++++++++++++++++++++++
 1 files changed, 36 insertions(+), 0 deletions(-)

diff --git a/gst/camerabin/gstcamerabin.c b/gst/camerabin/gstcamerabin.c
index 11f8681..fc9b44e 100644
--- a/gst/camerabin/gstcamerabin.c
+++ b/gst/camerabin/gstcamerabin.c
@@ -2830,6 +2830,32 @@ gst_camerabin_zoom_notify_cb (GObject * video_source, GParamSpec * pspec,
 }
 
 /*
+ * gst_camerabin_preview_notify_cb:
+ * @video_source: videosrc object
+ * @pspec:        GParamSpec for property
+ * @user_data:    camerabin object
+ *
+ * Set preview caps to video source if they're configured to camerabin,
+ * but not in use in video source.
+ *
+ */
+static void
+gst_camerabin_preview_notify_cb (GObject * video_source, GParamSpec * pspec,
+    gpointer user_data)
+{
+  GstCameraBin *camera = GST_CAMERABIN (user_data);
+  if (gst_photography_get_format (GST_PHOTOGRAPHY (camera->src_vid_src),
+          GST_PHOTOGRAPHY_OPERATION_MODE_PREVIEW) == NULL
+      && camera->preview_caps) {
+    /* Preview caps have been set to camerabin, but they are not in effect
+       in video source. This may happen if preview caps were previously set
+       while video source was in NULL state or video source has reset them. */
+    GST_DEBUG_OBJECT (camera, "setting preview caps again");
+    gst_camerabin_set_video_source_preview_caps (camera);
+  }
+}
+
+/*
  * gst_camerabin_monitor_video_source_properties:
  * @camera: camerabin object
  *
@@ -2859,6 +2885,14 @@ gst_camerabin_monitor_video_source_properties (GstCameraBin * camera)
         camera->src_vid_src);
     g_signal_connect (G_OBJECT (camera->src_vid_src), "notify::zoom",
         (GCallback) gst_camerabin_zoom_notify_cb, camera);
+    GST_DEBUG_OBJECT (camera,
+        "connecting to %" GST_PTR_FORMAT
+        " - notify::image-preview-supported-caps", camera->src_vid_src);
+    /* Use _connect_after() so that application has the priority for 
+       handling the notification, we check it only as fallback */
+    g_signal_connect_after (G_OBJECT (camera->src_vid_src),
+        "notify::image-preview-supported-caps",
+        (GCallback) gst_camerabin_preview_notify_cb, camera);
   }
 }
 
@@ -4133,6 +4167,8 @@ gst_camerabin_change_state (GstElement * element, GstStateChange transition)
           gst_camerabin_scene_mode_notify_cb, camera);
       g_signal_handlers_disconnect_by_func (camera->src_vid_src,
           gst_camerabin_zoom_notify_cb, camera);
+      g_signal_handlers_disconnect_by_func (camera->src_vid_src,
+          gst_camerabin_preview_notify_cb, camera);
 
       /* we might need to update our caps when going back to PLAYING */
       camera->image_capture_caps_update = TRUE;
