diff --git a/gst-libs/gst/interfaces/photography.h b/gst-libs/gst/interfaces/photography.h
index 0475d71aa..5352ba868 100644
--- a/gst-libs/gst/interfaces/photography.h
+++ b/gst-libs/gst/interfaces/photography.h
@@ -25,7 +25,7 @@
 
 #ifndef GST_USE_UNSTABLE_API
 #warning "The GstPhotography interface is unstable API and may change in future."
-#warning "You can define GST_USE_UNSTABLE_API to avoid this warning." 
+#warning "You can define GST_USE_UNSTABLE_API to avoid this warning."
 #endif
 
 #include <gst/gst.h>
@@ -215,6 +215,10 @@ typedef enum
  * @GST_PHOTOGRAPHY_COLOR_TONE_MODE_WHITEBOARD: Whiteboard (Since 1.2)
  * @GST_PHOTOGRAPHY_COLOR_TONE_MODE_BLACKBOARD: Blackboard (Since 1.2)
  * @GST_PHOTOGRAPHY_COLOR_TONE_MODE_AQUA: Aqua (Since 1.2)
+ * @GST_PHOTOGRAPHY_COLOR_TONE_MODE_EMBOSS: Aqua (Since 1.?)
+ * @GST_PHOTOGRAPHY_COLOR_TONE_MODE_SKETCH: Aqua (Since 1.?)
+ * @GST_PHOTOGRAPHY_COLOR_TONE_MODE_NEON: Aqua (Since 1.?)
+ *
  *
  * Modes for special color effects.
  */
@@ -235,7 +239,10 @@ typedef enum
   GST_PHOTOGRAPHY_COLOR_TONE_MODE_POSTERIZE,
   GST_PHOTOGRAPHY_COLOR_TONE_MODE_WHITEBOARD,
   GST_PHOTOGRAPHY_COLOR_TONE_MODE_BLACKBOARD,
-  GST_PHOTOGRAPHY_COLOR_TONE_MODE_AQUA
+  GST_PHOTOGRAPHY_COLOR_TONE_MODE_AQUA,
+  GST_PHOTOGRAPHY_COLOR_TONE_MODE_EMBOSS,
+  GST_PHOTOGRAPHY_COLOR_TONE_MODE_SKETCH,
+  GST_PHOTOGRAPHY_COLOR_TONE_MODE_NEON
 } GstPhotographyColorToneMode;
 
 /**
