diff --git a/gst-libs/gst/interfaces/photography.h b/gst-libs/gst/interfaces/photography.h
index 0475d71aa..56537cb0b 100644
--- a/gst-libs/gst/interfaces/photography.h
+++ b/gst-libs/gst/interfaces/photography.h
@@ -264,6 +264,12 @@ typedef enum
  *     of scenes lit by candles (Since 1.2)
  * @GST_PHOTOGRAPHY_SCENE_MODE_BARCODE: Applications are looking for
  *     a barcode (Since 1.2)
+ * @GST_PHOTOGRAPHY_SCENE_MODE_ASD: Automatic Scene Detection (Since 1.?)
+ * @GST_PHOTOGRAPHY_SCENE_MODE_BACKLIGHT: Backlit photos (Since 1.?)
+ * @GST_PHOTOGRAPHY_SCENE_MODE_FLOWERS: For shooting flowers (Since 1.?)
+ * @GST_PHOTOGRAPHY_SCENE_MODE_AR: Specific for Augmented Reality (Since 1.?)
+ * @GST_PHOTOGRAPHY_SCENE_MODE_HDR: High Dynamic Range photography (Since 1.?)
+ *
  *
  * Each mode contains preset #GstPhotography options in order to produce
  * good capturing result in certain scene.
@@ -287,7 +293,12 @@ typedef enum
   GST_PHOTOGRAPHY_SCENE_MODE_FIREWORKS,
   GST_PHOTOGRAPHY_SCENE_MODE_PARTY,
   GST_PHOTOGRAPHY_SCENE_MODE_CANDLELIGHT,
-  GST_PHOTOGRAPHY_SCENE_MODE_BARCODE
+  GST_PHOTOGRAPHY_SCENE_MODE_BARCODE,
+  GST_PHOTOGRAPHY_SCENE_MODE_ASD,
+  GST_PHOTOGRAPHY_SCENE_MODE_BACKLIGHT,
+  GST_PHOTOGRAPHY_SCENE_MODE_FLOWERS,
+  GST_PHOTOGRAPHY_SCENE_MODE_AR,
+  GST_PHOTOGRAPHY_SCENE_MODE_HDR
 } GstPhotographySceneMode;

 /**
